scripts/build-job-matrix.py

```diff
@@ -0,0 +1,62 @@
+import json
+import os
+import sys
+from itertools import product
+
+def load_config(config_path, default_config=None):
+    """Load JSON configuration from the given path."""
+    try:
+        with open(config_path, 'r') as file:
+            config = json.load(file)
+            if not config and default_config is not None:
+                return default_config
+            return config
+    except (FileNotFoundError, json.JSONDecodeError):
+        if default_config is not None:
+            return default_config
+        else:
+            raise
+
+def main(image_dirs):
+    print("Image directories:", image_dirs)  
+
+    default_config_path = 'opensciencegrid/default-build-config.json'
```

Comment:
Nitpick: this should be a top level constant of the library. Like inserting this between the last import and your first function definition:

```
DEFAULT_CONFIG_PATH = 'opensciencegrid/default-build-config.json'
```